test/package.js

```diff
@@ -21,3 +25,16 @@ test('Every rule is defined in index file', async t => {
 	t.is(Object.keys(index.configs.recommended.rules).length, ruleFiles.length,
 		'There are more exported rules in the recommended config than rule files.');
 });
+
+test('Every rule has valid meta.type', t => {
+	const validTypes = ['problem', 'suggestion', 'layout'];
+
+	for (const file of ruleFiles) {
+		const name = path.basename(file, '.js');
+		const rule = index.rules[name];
+
+		t.true(rule.meta !== null && rule.meta !== undefined, `${name} has no meta`);
+		t.true(rule.meta.type !== null, `${name} has no meta.type`);
```

Comment:
This should also test again `undefined` too.

Reply:
Actually, it would be better to reverse it, and test against the expected value instead:
```suggestion
		t.is(typeof rule.meta.type, 'string', `${name} has no meta.type`);
```

Same with the above.

Reply:
Oh yeah, sorry, linter played a a bad joke with men. As of testing for type, it's tested to be one of values in the next line anyhow, (which includes same type).